lib/ElementHandle.js

```diff
@@ -295,6 +295,25 @@ class ElementHandle extends JSHandle {
     return result;
   }
 
+  /**
+   * @param {string} selector
+   * @param {Function|String} pageFunction
+   * @param {!Array<*>} args
+   * @return {!Promise<(!Object|undefined)>}
+   */
+  async $$eval(selector, pageFunction, ...args) {
```

Comment:
Let's re-use this implementation in `frame.$$eval` method similarly to how it's done for `farme.$eval`:

https://github.com/GoogleChrome/puppeteer/blob/17bc6515deec8229afb73edc3869f497e6c06e10/lib/FrameManager.js#L368-L372